File: src/main/java/io/lsdconsulting/lsd/distributed/diagram/event/SynchronousResponseBuilder.java
package io.lsdconsulting.lsd.distributed.diagram.event;

import com.lsd.IdGenerator;
import com.lsd.events.SequenceEvent;
import com.lsd.events.SynchronousResponse;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
public class SynchronousResponseBuilder implements SequenceEventBuilder {
    private final IdGenerator idGenerator;

    @Override
    public SequenceEvent build(String label, String serviceName, String target, String colour, String data) {
        return SynchronousResponse.builder()
                .id(idGenerator.next())
                .label(label)
                .from(target)
                .to(serviceName)
                .colour(colour)
                .data(data)
                .build();
    }
}

File: src/main/java/io/lsdconsulting/lsd/distributed/generator/diagram/event/builder/SynchronousResponseBuilder.kt
package io.lsdconsulting.lsd.distributed.generator.diagram.event.builder

import com.lsd.core.builders.MessageBuilder.Companion.messageBuilder
import com.lsd.core.domain.MessageType

fun buildSynchronousResponse(
    id: String,
    label: String,
    serviceName: String,
    target: String,
    colour: String,
    data: String
) = messageBuilder()
    .id(id)
    .type(MessageType.SYNCHRONOUS_RESPONSE)
    .label(label)
    .from(target)
    .to(serviceName)
    .colour(colour)
    .data(data)
    .build()
